Modules/_ctypes/_ctypes.c

```diff
@@ -5514,13 +5528,13 @@ Pointer_set_contents(PyObject *op, PyObject *value, void *closure)
 
     dst = (CDataObject *)value;
     if (dst != self) {
-        LOCK_PTR(dst);
+        Py_BEGIN_CRITICAL_SECTION(dst);
         locked_deref_assign(self, dst->b_ptr);
-        UNLOCK_PTR(dst);
+        Py_END_CRITICAL_SECTION();
     } else {
-        LOCK_PTR(self);
+        Py_BEGIN_CRITICAL_SECTION(self);
         *((void **)self->b_ptr) = dst->b_ptr;
```

Comment:
Can you explain again why we can't use `locked_deref_assign` here? is it because of pointer rules? (because here, we're just re-assigning the stuff right?)

Reply:
I would like to get rid of locked_deref_assign  in future so using it less if better. Also that reacquires the critical sections so will cause perf loss